### PR TITLE
[Core] Fixing process

### PR DESCRIPTION
**Pull request by GuillermoCasas**
This process does not work consistently (it compares the analytical solution at t=current_time) with the solution at t=current_time-delta_time, although the check is done after solving the system. It was making my own tests from SwimmingDEM fail. Related to #4413

### CODE REVIEW

**Review by loumalouomega — CHANGES_REQUESTED** [review]
I will block until I check the nightly tests on the Structural that uses this processes

**Comment by loumalouomega** [comment]
There is a test already failing in fact

File: applications/StructuralMechanicsApplication/tests/rigid_test/sphere_failing_parameters.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
                     "model_part_name" : "Structure.SelfWeight3D_Self_weight_Auto1",
                     "variable_name"   : "VELOCITY_Z",
                     "interval"        : [0.0, "End"],
-                    "value"           : "-9.81*(t + 0.05)",
+                    "value"           : "-9.81*t",
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/rigid_test/sphere_failing_parameters.json]: in the scope of this PR, this makes sense to me

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/rigid_test/sphere_failing_parameters.json]: Indeed

**Review by loumalouomega — DISMISSED** [review]

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**